vortex-array/flatbuffers/array.fbs

```diff
@@ -9,7 +11,22 @@ table Array {
     has_buffer: bool;
     encoding: uint16;
     metadata: [ubyte];
+    stats: ArrayStats;
     children: [Array];
 }
 
+table ArrayStats {
+    min: vortex.scalar.Scalar;
```

Comment:
We use scalars in memory for stats mostly because we're a little lazy, but also because it gives us a lightweight wrapper around runtime-erased types. (The alternative would be like some StatsValue enum of `bool | usize | Scalar`)

Anyway, in the serialised form we may as well use strong types where we can. So min/max have to remain as scalars, but is_sorted can be a bool, run_count can be a u64, bit_width_freq can be a `[u64]` etc.

Reply:
WYDT @robert3005 ?

Reply:
Makes sense imho. Probably avoids some overhead in the serialised message